File: leetcode/0167.Two-Sum-II-Input-Array-Is-Sorted/TwoSum2InputArrayIsSorted_test.go
package leetcode

import (
	"reflect"
	"testing"
)

func Test(t *testing.T) {
	testSuits := []struct {
		name   string
		nums   []int
		target int
		ans    []int
	}{
		{"Example 1", []int{2, 7, 11, 15}, 9, []int{1, 2}},
	}
	for _, test := range testSuits {
		got := twoSum(test.nums, test.target)

		if !reflect.DeepEqual(got, test.ans) {
			t.Errorf("[%v] expect '%v', got '%v' ", test.name, test.ans, got)
		}
	}
}

File: leetcode/0167.Two-Sum-II-Input-Array-Is-Sorted/TwoSum2InputArrayIsSorted_test.py

from TwoSum2InputArrayIsSorted import Solution
import pytest
from copy import deepcopy

SOL = Solution()

TEST_SUITS =  [
    ([2,7,11,15], 9, [1, 2]),
]

@pytest.mark.parametrize(
    "nums, target, ans", 
    deepcopy(TEST_SUITS)
)
def test(nums, target, ans):
    assert SOL.twoSum(nums, target) == ans